### PR TITLE
Document the `onKeyChange` prop for custom widgets

### PR DESCRIPTION
**Pull request by rafegoldberg**
### Reasoning

In order to render OAS/Swagger compliant specs (specifically objects with `additionalProperties`) we needed to support customizable key names in our custom field template. I had to poke through the rendered React component in the console to discover the `onKeyChange` handler, as [no mention made of it was made](https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/#custom-widget-components) in the relevant docs!

### Checklist

* [x] **I'm updating documentation**
  - [x] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added

### CODE REVIEW

**Comment by epicfaace** [comment]
Thanks! Can you add language clarifying that `onKeyChange` is only practically used with additionalProperties?

**Comment by rafegoldberg** [comment]
@epicfaace add language around `additionalProperties`; let me know if this looks good.

File: packages/core/docs/advanced-customization.md [hunk 1]
```diff
@@ -354,6 +354,7 @@ The following props are passed to custom widget components:
 - `readonly`: `true` if the widget is read-only;
 - `autofocus`: `true` if the widget should autofocus;
 - `onChange`: The value change event handler; call it with the new value everytime it changes;
+- `onKeyChange`: The key change event handler for fields with `additionalProperties`; pass the new value everytime it changes;
```
Comment (epicfaace) [inline packages/core/docs/advanced-customization.md]:
```suggestion
- `onKeyChange`: The key change event handler (only called for fields with `additionalProperties`); pass the new value everytime it changes;
```

**Comment by epicfaace** [comment]
@rafegoldberg thanks! Made a small suggestion -- could you accept it?

**Pull request merged**